src/Parser/FrostMage/Modules/Features/IcyVeinsDuration.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Module from 'Parser/Core/Module';
+
+class IcyVeinsDuration extends Module {
+  static dependencies = {
+		combatants: Combatants,
+	}
+	
+  on_initialized() {
+	const hasThermalVoid = this.combatants.selected.hasTalent(SPELLS.THERMAL_VOID_TALENT.id);
+	this.active = hasThermalVoid;
+  }
+
+  IcyVeinsCasts = 0;
+	
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.ICY_VEINS.id) {
+      return;
+    }
+    this.IcyVeinsCasts += 1;
+  }
+
+  suggestions(when) {
+    const IcyVeinsUptime = (this.combatants.selected.getBuffUptime(SPELLS.ICY_VEINS.id) / 1000) / this.IcyVeinsCasts;
```

Comment:
I'd say "uptime" isn't accurate depiction of what is calculated here. "averageDuration" (since we're talking about Icy Veins it's unnecessary to repeat) would be better.

Reply:
makes sense, ill change it.